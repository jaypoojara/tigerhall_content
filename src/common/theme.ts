export const theme = {
  palette: {
    teal: {
      1000: "#001315",
    },
    grey: {
      500: "#BBB8B2",
      700: "#797670",
      800: "#4D4B46",
      900: "#383733",
    },
    orange: {
      50: "#FFF9F6",
      400: "#FFA97A",
      600: "#FF5900",
    },
    white: "#FFFFFF",
    black: "#000000",
  },
};
