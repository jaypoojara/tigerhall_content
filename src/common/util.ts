import { IContentCard } from "../components/ContentCard/type";
import { IContentData } from "../service/graphql/responseType";

export const mapContentCardsData = (data: IContentData): IContentCard => {
  return {
    name: data.name,
    image: data.image.uri.replace('https://images.staging.tigerhall.io/', 'https://images.staging.tigerhall.io/resize/250x/'),
    author: data.experts?.[0].firstName + data.experts?.[0].lastName,
    subtext: "COMMUNICATING AS A LEADER",
    completed: "30% Completed",
    duration: "20m",
    org: "Subway APAC",
  };
};
