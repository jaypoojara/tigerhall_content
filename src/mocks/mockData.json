{
  "contentCards": {
    "edges": [
      {
        "name": "test qa",
        "image": {
          "uri": "https://images.staging.tigerhall.io/users/2023-06-14/3f6c372d-d0b3-42fc-87f5-ffea83c6c5fe-f529ae53-39e2-4dea-861b-47cc9ceafeae.png",
          "__typename": "Image"
        },
        "categories": [
          {
            "name": "category April",
            "__typename": "Category"
          }
        ],
        "experts": [
          {
            "firstName": "Aqib",
            "lastName": "Abas",
            "title": "",
            "company": "",
            "__typename": "Expert"
          }
        ],
        "__typename": "Podcast"
      }
    ],
    "__typename": "ContentConnection"
  }
}
