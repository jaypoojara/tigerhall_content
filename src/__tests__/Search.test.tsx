import { fireEvent, render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import SearchPage from "../pages/Search";
import { ThemeProvider } from "styled-components";
import { useQuery } from "@apollo/client";
import { theme } from "../common/theme";
import mockData from '../mocks/mockData.json';
import searchedMockData from '../mocks/searchedMockData.json';
jest.mock("@apollo/client");

describe("SearchPage", () => {
  it("should render content card data", async () => {
    (useQuery as jest.Mock).mockReturnValueOnce({
      loading: false,
      error: null,
      data: mockData,
    });
    render(
      <MockedProvider>
        <ThemeProvider theme={theme}>
          <SearchPage />
        </ThemeProvider>
      </MockedProvider>
    );
    const defaultName = "test qa";
    expect(screen.getByText(defaultName)).toBeInTheDocument();

  });

  it("should render searched content card data", async () => {
    (useQuery as jest.Mock).mockReturnValueOnce({
      loading: false,
      error: null,
      data: mockData,
    });
    render(
      <MockedProvider>
        <ThemeProvider theme={theme}>
          <SearchPage />
        </ThemeProvider>
      </MockedProvider>
    );
    const defaultName = "test developer";
    const searchTerm = "test qa";
    expect(screen.getByText(searchTerm)).toBeInTheDocument();

    const searchInput = screen.getByTestId("searchInput");
    
    (useQuery as jest.Mock).mockReturnValueOnce({
      loading: false,
      error: null,
      data: searchedMockData,
    });
    fireEvent.change(searchInput, { target: { value: "test developers" } });
    fireEvent.keyDown(searchInput, { key: "Enter", code: "Enter" });
    expect(screen.getByText(defaultName)).toBeInTheDocument();
  });
});
