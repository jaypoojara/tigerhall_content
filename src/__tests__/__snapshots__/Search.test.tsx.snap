// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should match snapshot 1`] = `
<div>
  <div
    class="sc-kstpWv kXWKKy"
  >
    <div
      class="sc-hBEZvw dIxoLP"
    >
      <div
        class="sc-fodVRF jPWXmj"
      >
        <img
          alt="SearchIcon"
          src="SearchIcon.svg"
        />
        <input
          class="sc-fFubCH fVwCXN"
          data-testid="searchInput"
        />
      </div>
      <div
        class="sc-bdfCDU kGOMDk"
      >
        <div
          class="sc-gsTDqH dzvPpc"
          imageurl="https://images.staging.tigerhall.io/resize/250x/users/2023-06-14/3f6c372d-d0b3-42fc-87f5-ffea83c6c5fe-f529ae53-39e2-4dea-861b-47cc9ceafeae.png"
        >
          <div
            class="sc-dlfmHC iPRmzj"
          >
            <img
              alt="ContentStatusIcon"
              src="PartiallyCompleteClock.svg"
            />
            <div
              class="sc-jrAGKZ sc-iqHXzD ibXuTY gKFJEZ"
            >
              30% Completed
            </div>
          </div>
          <div
            class="sc-hKgKIp edEyJb"
          >
            <img
              alt="EllipseIcon"
              src="Ellipse.svg"
            />
            <img
              alt="PodcastIcon"
              src="PodcastIcon.svg"
            />
          </div>
          <div
            class="sc-eCstZk gcvkBc"
          >
            <div
              class="sc-bqyKbq ikmMeJ"
            >
              <img
                alt="TimeIcon"
                src="TimeIcon.svg"
              />
              <div
                class="sc-jrAGKZ sc-iqHXzD ibXuTY gZHZOe"
                color="#FFFFFF"
              >
                20m
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jSguLX iYfvCp"
        >
          <img
            alt="ProgressBarIcon"
            src="ProgressBarIcon.svg"
          />
        </div>
        <div
          class="sc-iBPTVF cncdJn"
        >
          <div
            class="sc-gKsecS dTZiMq"
          >
            <div
              style="display: flex; flex-direction: column; gap: 2px;"
            >
              <div
                class="sc-jrAGKZ sc-crrtmM ibXuTY dbuhjB"
              >
                COMMUNICATING AS A LEADER
              </div>
              <div
                class="sc-kEjbdu sc-dQpIV gwGjDr hfUQze"
              >
                test qa
              </div>
            </div>
            <div>
              <div
                class="sc-jrAGKZ sc-crrtmM ibXuTY dbuhjB"
              >
                AqibAbas
              </div>
              <div
                class="sc-jrAGKZ sc-iqHXzD ibXuTY gKFJEZ"
              >
                Subway APAC
              </div>
            </div>
          </div>
          <div
            class="sc-fubDmA iDUUgp"
          >
            <img
              alt="ShareIcon"
              src="ShareIcon.svg"
            />
            <img
              alt="BookmarkIcon"
              src="BookmarkIcon.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
